Add web API entry to edit trial status

notoraptor opened the pull request:
# Description

Hi @bouthilx ! This is a PR to edit trial status from web API in backend. No yet changes in dashboard.

# Changes

- Add web API entry to edit trial status: `/trials/:experiment/:id/set-status/:status`
- Add doc for new web API entry
- Add unit tests

# Checklist

## Tests
- [ ] I added corresponding tests for bug fixes and new features. If possible, the tests fail without the changes
- [ ] All new and existing tests are passing (`$ tox -e py38`; replace `38` by your Python version if necessary)

## Documentation
- [ ] I have updated the relevant documentation related to my changes

## Quality
- [ ] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [ ] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [ ] My code follows the style guidelines (`$ tox -e lint`)

# Further comments
_Please include any additional information or comment that you feel will be helpful to the review of this pull request._

notoraptor committed:
Add web API entry to edit trial status: `/trials/:experiment/:id/set-status/:status`

bouthilx commented on tests/functional/serving/test_trials_resource.py:
```diff
@@ -343,3 +343,52 @@ def test_get_trial(self, client, ephemeral_storage):
             "statistics": {"a": 10, "b": 5},
             "status": "completed",
         }
+
```
Should test what happens when setting status to non-existing trial in existing experiment, and non-existing trial in non-existing experiment.

notoraptor replied:
Done! https://github.com/Epistimio/orion/pull/1092/commits/cdc75251a441da905aaa3184a580440c686ec927

notoraptor committed:
Test what happens when setting status to non-existing trial in existing experiment, and non-existing trial in non-existing experiment.

notoraptor committed:
[ci][playwright tests]
- Redirect both stdout and stderr output from orion server to a file
- Update actions/setup=python from v2 to v4
- Check ulimit

[dashboard][tests]
Sub-bar in progress bar may have width 0, thus be considered as visible.
Wait for such bar with status "attached" instead of "visible" so that playwright accepts to find it.

notoraptor commented:
Hello @bouthilx ! I updated this PR:
- I removed debug printing
- I kept small changes that were introduced to try fixing CI failures, and I think they are still useful: all in last commit: https://github.com/Epistimio/orion/pull/1092/commits/c6bcf9648821db9f6ba908d81b4c96b33feee780

Playwright tests succeed! So I guess you can make a final review here!
